### PR TITLE
Reservation list

### LINKED ISSUE #19: Reservation overview implementation
This refers to the view where all reservations associated with a given event are listed.

Base implementation:

- [x] Reservation List (using mock data)
- [x] Search functionality

Full implementation:

- [ ] Load list information from API
- [ ] Implement the various button actions (requires #9 )

### PR DESCRIPTION
**Pull request by pimber**

### CODE REVIEW

File: Flea/Pages/Booking.razor [hunk 1]
```diff
@@ -1,6 +1,98 @@
...
+    
+    private List<Reservation> _reservations = new()
+    {
+        {new Reservation() {Name = "Sebastian Aaholm", Stande = 3, PaymentStatus = false}},
+        {new Reservation() {Name = "Jens Jensen", Stande = 3, PaymentStatus = true}},
+        {new Reservation() {Name = "Hans Nielsen", Stande = 10, PaymentStatus = false}},
+    };
+    // ---
+    
+    private List<Reservation> _shownReservation = new();
```
Comment (jonasKjellerup) [inline Flea/Pages/Booking.razor]: The type here can be made more general (`IEnumerable<Reservation>`) allowing for simplification in the search function.

File: Flea/Pages/Booking.razor [hunk 2]
```diff
@@ -1,6 +1,98 @@
...
+    class Reservation
+    {
+        public string Name { get; set; }
+        public int Stande { get; set; }
+        public bool PaymentStatus { get; set; }
+    }
+    
+    private List<Reservation> _reservations = new()
+    {
+        {new Reservation() {Name = "Sebastian Aaholm", Stande = 3, PaymentStatus = false}},
```
Comment (jonasKjellerup) [inline Flea/Pages/Booking.razor]: You don't need the second set of braces.
```cs
new Reservation {Name = "Sebastian Aaholm", Stande = 3, PaymentStatus = false},
```
is sufficient

File: Flea/Pages/Booking.razor [hunk 3]
```diff
@@ -1,6 +1,98 @@
...
+    }
+</div>
+
+@code
+{
+    // Dummy data ---
+    class Reservation
+    {
+        public string Name { get; set; }
+        public int Stande { get; set; }
```
Comment (jonasKjellerup) [inline Flea/Pages/Booking.razor]: Avoid using danish words in identifier names. (I am aware that it is merely placeholder but try keep it in mind still)

File: Flea/Pages/Booking.razor [hunk 4]
```diff
@@ -1,6 +1,98 @@
...
+            _shownReservation = _reservations;
+        }
+        else if (Char.IsDigit(input[0]))
+        {// Finds by stande
+
+            var results = from c
+                in _reservations
+                where c.Stande.ToString().Contains(input)
+                select c;
+            _shownReservation= results.ToList();
```
Comment (jonasKjellerup) [inline Flea/Pages/Booking.razor]: By implementing the aforementioned change to `_shownReservation` (changing its type to `IEnumerable<Reservation>`) you can simplify this by omitting the intermediary variable and assigning directly to `_shownReservation`.

```cs
_shownReservation = from c
                in _reservations
                where c.Stande.ToString().Contains(input)
                select c;
```

File: Flea/Pages/Booking.razor [hunk 5]
```diff
@@ -1,6 +1,98 @@
...
+        }
+        else if (input.Contains("betalt"))
```
Comment (jonasKjellerup) [inline Flea/Pages/Booking.razor]: This case doesn't really make sense unless we also support the inverse case ("ej betalt").

File: Flea/Shared/ReservationOverview.razor [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    <button class="button res-del-button" @onclick="Delete">Slet</button>
+    <button class="button res-edit-button" @onclick="Edit">Rediger</button>
+</div>
+
+@code {
+    [Parameter]
+    public string Name { get; set; }
+
+    [Parameter]
+    public int Stande { get; set; }
```
Comment (jonasKjellerup) [inline Flea/Shared/ReservationOverview.razor]: Again avoid danish words in source code identifiers.

File: Flea/Shared/ReservationOverview.razor [hunk 7]
```diff
@@ -0,0 +1,34 @@
+<div class="grid-container-res">
+    <div class="res-txt">@Name</div>
+    <div class="res-txt">@Stande</div>
+    <div class="res-txt">@(PaymentStatus ? "Betalt" : "Ej betalt")</div>
+    <button class="button res-del-button" @onclick="Delete">Slet</button>
+    <button class="button res-edit-button" @onclick="Edit">Rediger</button>
+</div>
+
```
Comment (jonasKjellerup) [inline Flea/Shared/ReservationOverview.razor]: The file/component name is slightly misleading as it implies that component implements an overview of reservations, however this is in actuality done by the `Booking` view. This component merely an item in a overview reservation.

**Review by jonasKjellerup — CHANGES_REQUESTED** [review]
Another more general thing that I think needs to be discussed before merging any more UI related PRs is the approach to styling. We originally discussed trying to make it look as close to native IOS components as possible, hence the inclusion of OnsenUI, however we seem to deliberately be working in opposition to this idea in many cases (e.g the buttons in this commit are using custom CSS override the OnsenUI CSS).

File: Flea/Shared/ReservationOverview.razor (hunk 7 above)
Comment (jonasKjellerup) [reply Flea/Shared/ReservationOverview.razor]: This seems to have been solved in 446f8e0

File: Flea/Pages/Booking.razor (hunk 1 above)
Comment (pimber) [reply Flea/Pages/Booking.razor]: Fixed

File: Flea/Pages/Booking.razor (hunk 2 above)
Comment (pimber) [reply Flea/Pages/Booking.razor]: Noted

File: Flea/Pages/Booking.razor (hunk 3 above)
Comment (pimber) [reply Flea/Pages/Booking.razor]: Fixed

File: Flea/Pages/Booking.razor (hunk 4 above)
Comment (pimber) [reply Flea/Pages/Booking.razor]: Fixed thanks for help

File: Flea/Shared/ReservationOverview.razor (hunk 6 above)
Comment (pimber) [reply Flea/Shared/ReservationOverview.razor]: fixed

**Review by jonasKjellerup — APPROVED** [review]

**Pull request closed without merging**